### PR TITLE
Product Block Editor: adjust the attribute modal height based on attribute options

### LINKED ISSUE #47648: Product Block Editor: options dropdown overlaps the table row
Depending on the number of options the FormTokenField has or how long they are, the component overlaps the table row.

Screenshot 2024-05-21 at 10 01 39

Screenshot 2024-05-21 at 10 01 16

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR sets the attribute row height when the options dropdown height changes, fixing the issues described [here](https://github.com/woocommerce/woocommerce/issues/47648).

Closes https://github.com/woocommerce/woocommerce/issues/47648

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the product editor
2. Go to the Variations section
3. Create a new attribute
4. Create options for this attribute
5. Confirm that when the terms token field increases its height, the row gets increased too.

 

https://github.com/woocommerce/woocommerce/assets/77539/5664fdec-d6b6-4984-afcf-41730db8239e

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/attribute-control/attribute-table-row.tsx [hunk 1]
```diff
@@ -200,10 +200,40 @@ export const AttributeTableRow: React.FC< AttributeTableRowProps > = ( {
...
+		if ( ! tokenFieldElement || ! optionsWrapperElement ) {
+			return;
+		}
+
+		// Options wrapper height
+		const height = optionsWrapperElement.clientHeight;
+
+		// Set the height of the token field
+		tokenFieldElement.style.height = `${ height }px`;
+	}, [ selectedValues ] ); // need to know the height when the selected values change
```
Comment (louwie17) [inline packages/js/product-editor/src/components/attribute-control/attribute-table-row.tsx]: I did prefer if we could handle this with CSS, as this feels like a hack.

**Review by louwie17 — CHANGES_REQUESTED** [review]
See comment, preferring to handle/fix this with CSS versus a code change.

**Review by louwie17 — COMMENTED** [review]
Nice work, I do like this approach better, appreciate the updates!
I did notice one small issue which probably requires a max-width on the terms input field. Or on the modal.
attributes-modal-expanding

**Comment by retrofox** [comment]
> Nice work, I do like this approach better, appreciate the updates! I did notice one small issue which probably requires a max-width on the terms input field. Or on the modal. 

Nice one. I think it's fixed defining the modal size f6df5f1c86f946331a3e36ec727ef44238880470

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 Thanks for the updates @retrofox, this tested well and changes look good 👍

**Pull request merged**